src/WebJobs.Script/ScriptHostBuilderExtensions.cs

```diff
@@ -196,6 +195,8 @@ public static IHostBuilder AddScriptHostCore(this IHostBuilder builder, ScriptAp
                     services.TryAddEnumerable(ServiceDescriptor.Singleton<IHostedService, FunctionsScaleMonitorService>());
                 }
                 services.TryAddSingleton<FunctionsScaleManager>();
+
+                services.AddSingleton(s => s.GetRequiredService<IFunctionInvocationDispatcherFactory>().GetFunctionDispatcher());
```

Comment:
Avoid adding this as a dependency. FunctionInvocationDispatcherFactory is available. Use that instead